pkg/api/types.go

```diff
@@ -44,6 +44,7 @@ type Properties struct {
 	AgentPoolProfiles       []AgentPoolProfile      `json:"agentPoolProfiles"`
 	LinuxProfile            LinuxProfile            `json:"linuxProfile"`
 	WindowsProfile          WindowsProfile          `json:"windowsProfile"`
+	ExtensionsProfile       []ExtensionProfile      `json:"extensionsProfile"`
```

Comment:
Hi - we did add a check for this.  That is the role of supported-orchestrators.json.  That file contains a json array of supported orchestrators for the extension.  There is a function orchestratorSupportsExtension that validates that the configured orchestrator is in that array.